src/acceptance-test/java/net/consensys/orion/acceptance/send/receive/privacyGroup/AddToPrivacyGroupTest.java

```diff
@@ -0,0 +1,71 @@
+ */
+package net.consensys.orion.acceptance.send.receive.privacyGroup;
+
+import static net.consensys.orion.acceptance.NodeUtils.createPrivacyGroupTransaction;
+import static net.consensys.orion.acceptance.NodeUtils.findPrivacyGroupTransaction;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import net.consensys.cava.junit.TempDirectoryExtension;
+import net.consensys.orion.acceptance.EthClientStub;
+import net.consensys.orion.acceptance.NodeUtils;
+import net.consensys.orion.acceptance.WaitUtils;
+import net.consensys.orion.http.handler.privacy.PrivacyGroup;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * Runs up a three nodes that communicates with each other and tries to add the third node to an existing privacy group.
+ */
+@ExtendWith(TempDirectoryExtension.class)
+class AddToPrivacyGroupTest extends PrivacyGroupAcceptanceTest {
+
+  @Test
+  void addToPrivacyGroup() {
+    final EthClientStub firstNode = NodeUtils.client(firstOrionLauncher.clientPort(), firstHttpClient);
+    final EthClientStub secondNode = NodeUtils.client(secondOrionLauncher.clientPort(), secondHttpClient);
+    final EthClientStub thirdNode = NodeUtils.client(thirdOrionLauncher.clientPort(), thirdHttpClient);
+
+    final String name = "testName";
+    final String description = "testDescription";
+    final String[] addresses = new String[] {PK_1_B_64, PK_2_B_64};
+    // create a privacy group
+    final PrivacyGroup privacyGroup = createPrivacyGroupTransaction(firstNode, addresses, PK_1_B_64, name, description);
+
+    final String privacyGroupId = privacyGroup.getPrivacyGroupId();
+    assertEquals(name, privacyGroup.getName());
+    assertEquals(description, privacyGroup.getDescription());
+    assertEquals(2, privacyGroup.getMembers().length);
+
+    NodeUtils.addToPrivacyGroup(firstNode, PK_3_B_64, PK_1_B_64, privacyGroupId);
+
+    final String[] newGroupMembers = new String[] {PK_1_B_64, PK_2_B_64, PK_3_B_64};
+
+    WaitUtils.waitFor(() -> assertEquals(1, findPrivacyGroupTransaction(secondNode, newGroupMembers).length));
+
+    final PrivacyGroup[] propagatedToExistingNode = findPrivacyGroupTransaction(secondNode, newGroupMembers);
+    assertArrayEquals(newGroupMembers, propagatedToExistingNode[0].getMembers());
+    assertEquals(privacyGroupId, propagatedToExistingNode[0].getPrivacyGroupId());
+
+    final PrivacyGroup[] propagatedToOriginalNode = findPrivacyGroupTransaction(firstNode, newGroupMembers);
+    assertArrayEquals(newGroupMembers, propagatedToOriginalNode[0].getMembers());
+    assertEquals(privacyGroupId, propagatedToOriginalNode[0].getPrivacyGroupId());
+
+    final PrivacyGroup[] propagatedToNewNode = findPrivacyGroupTransaction(thirdNode, newGroupMembers);
+    assertArrayEquals(newGroupMembers, propagatedToNewNode[0].getMembers());
+    assertEquals(privacyGroupId, propagatedToNewNode[0].getPrivacyGroupId());
+  }
+
```

Comment:
Idea (maybe big, maybe not useful): Can you create an acceptanceTest which shows that a remote Orion node cannot inject an "addd(me, group)" via the nodePort? I.e. effectively a byzantine node.